src/coreclr/jit/compiler.h

```diff
@@ -8873,6 +8891,42 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     }
 #endif // FEATURE_SIMD
 
+    var_types roundDownMaxRegSize(unsigned size)
+    {
+        assert(size > 0);
+        ssize_t maxRegSize = TARGET_POINTER_SIZE;
+#if defined(FEATURE_SIMD) && defined(TARGET_XARCH)
+        if (IsBaselineSimdIsaSupported())
+        {
+            maxRegSize = max(maxRegSize, (ssize_t)roundDownSIMDSize((unsigned)size));
+        }
+#endif
+        int nearestPow2 = 1 << BitOperations::Log2((unsigned)size);
+        switch (min((int)maxRegSize, nearestPow2))
+        {
+            case 1:
+                return TYP_UBYTE;
+            case 2:
+                return TYP_USHORT;
+            case 4:
+                return TYP_INT;
+            case 8:
+                return TYP_LONG;
+#ifdef FEATURE_SIMD
+            case 16:
+                return TYP_SIMD16;
+#ifdef TARGET_XARCH
+            case 32:
+                return TYP_SIMD32;
+            case 64:
+                return TYP_SIMD64;
+#endif
+#endif
```

Comment:
A question (I also have some similar logic in #85323) -- is it ok to create nodes with these SIMD16/32/64 types without checking for support for these types in some way?

Reply:
In this case support is checked, this method invokes `roundDownSIMDSize` that does that